internal/providers/github/github_rest.go

```diff
@@ -623,3 +644,19 @@ func (c *RestClient) ListEmails(ctx context.Context, opts *github.ListOptions) (
 	}
 	return emails, nil
 }
+
+func performWithRetry(ctx context.Context, op func() error) error {
+	exponentialBackOff := backoffv4.NewExponentialBackOff()
+	maxRetriesBackoff := backoffv4.WithMaxRetries(exponentialBackOff, MaxRateLimitRetries)
+	return backoffv4.Retry(op, backoffv4.WithContext(maxRetriesBackoff, ctx))
+}
+
+func isRateLimitError(err error) bool {
+	var rateLimitError *github.RateLimitError
+	isRateLimitErr := errors.As(err, &rateLimitError)
+
+	var abuseRateLimitError *github.AbuseRateLimitError
+	isAbuseRateLimitErr := errors.As(err, &abuseRateLimitError)
+
+	return isRateLimitErr || isAbuseRateLimitErr
```

Comment:
can we just do:
```go
func isRateLimitError(err error) bool {
    var rateLimitError *github.RateLimitError
    var abuseRateLimitError *github.AbuseRateLimitError
    return errors.As(err, &rateLimitError) || errors.As(err, &abuseRateLimitError)
}
```

Reply:
although on second though, the extra boolean flags might be useful for extra debugging (I'll add another comment with better locality about that)